src/canopy/knowledge_base/record_encoder/hybrid.py

```diff
@@ -0,0 +1,122 @@
+from typing import List
+from functools import cached_property
+
+from pinecone_text.dense import OpenAIEncoder
+from pinecone_text.dense.base_dense_ecoder import BaseDenseEncoder
+from pinecone_text.hybrid import hybrid_convex_scale
+from pinecone_text.sparse import BM25Encoder
```

Comment:
I agree in general if we assume pinecone_text.BaseDenseEncoder & pinecone_text.BM25Encoder exists. I tried replacing but it causes super long paths that are duplicating making the code much less readable. The modules are not exposed properly. I would suggest to wait for a clean API (if we will keep supporting pinecone-text) before changing it to pinecone-text.XXX I think in any case we will move pinecone-text inside the library, then it might be much easier to make some changes.